### PR TITLE
feat: Allow setting project type during creation

### LINKED ISSUE #1961: Create a training project
During project creation, it's not possible to select if it should be a training or project: 

image

The option should be added, analogous to the "Modify project" settings: 
image

### PR DESCRIPTION
**Pull request by zusorio**
Closes #1961

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -80,5 +80,6 @@
     "tailwindcss": "^3.4.14",
     "typescript": "^5.5.4",
     "typescript-eslint": "8.12.2"
-  }
+  },
+  "packageManager": "pnpm@9.11.0+sha512.0a203ffaed5a3f63242cd064c8fb5892366c103e328079318f78062f24ea8c9d50bc6a47aa3567cabefd824d170e78fa2745ed1f16b132e16436146b7688f19b"
```
Comment (MoritzWeber0) [inline frontend/package.json:84]: Was this change intended?

File: frontend/src/app/projects/create-project/create-project.component.html [hunk 2]
```diff
@@ -66,6 +66,21 @@ <h3 class="flex justify-center sm:hidden">Create new project</h3>
                 </mat-radio-group>
               </div>
             </fieldset>
+            <fieldset>
+              <div class="flex flex-col">
+                <legend>Project type</legend>
+                <mat-radio-group formControlName="type" class="flex flex-col">
+                  <mat-radio-button
+                    *ngFor="
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/create-project/create-project.component.html:74]: Please use the new control flow: https://angular.dev/guide/templates/control-flow#repeat-content-with-the-for-block

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 1 above)
Comment (zusorio) [reply frontend/package.json:84]: Nope, corepack randomly adds that and i forgot to remove it this time. I need to find a way to turn that off at some point.

File: frontend/src/app/projects/create-project/create-project.component.html [hunk 3]
```diff
@@ -53,16 +53,33 @@ <h3 class="flex justify-center sm:hidden">Create new project</h3>
...
+                        )
+                      }}
+                    </mat-radio-button>
+                  }
+                </mat-radio-group>
+              </div>
+            </fieldset>
+            <fieldset>
+              <div class="flex flex-col">
+                <legend>Project type</legend>
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/create-project/create-project.component.html:73]: image

The gap between the title and the radios looks a bit compressed. This is due to the description of General in the brackets, which expands to two lines. 

I'd either reduce the length of the description, e. g., by removing the "multiple" word. Or alternatively add the description always in a new line with smaller text and dark grey color.

File: frontend/src/storybook/project.ts [hunk 4]
```diff
@@ -51,6 +52,14 @@ export class MockProjectWrapperService
   getAvailableVisibilities(): Visibility[] {
     return Object.keys(ProjectVisibilityDescriptions) as Visibility[];
   }
+
+  getProjectTypeDescription(type: ProjectType): string {
```
Comment (MoritzWeber0) [inline frontend/src/storybook/project.ts:56]:
```suggestion
  get projectTypeDescription(type: ProjectType): string {
```

You can use getter for it and the next function.

**Review by MoritzWeber0 — APPROVED** [review]
